docs: Add message formats to contributing.md (#176)
allinbits/cosmos-cash: TECHNICAL-SETUP.md

# Technical setup 

## Software requirement

To build the project:

- [Golang](https://golang.org/dl/) v1.16 or greater 
- [`make`](https://www.gnu.org/software/make/) - to use Makefile targets
- `sha256sum` = to calculate build checksums

To run the [seed scripts](scripts/seeds/README.md)

- OS: linux or macOS
- [jq](https://stedolan.github.io/jq/) - json processor

For git pre-commit hooks

- python
- [pre-commit](https://pre-commit.com/#install) - git pre-commit hooks framework 
- [golangci-lint](https://github.com/golangci/golangci-lint) - linter for Go 

### Swagger 

> To generate the swagger UI the following dependencies must be installed
> 
> - [`swagger-combine`](https://www.npmjs.com/package/swagger-combine)

Reviewer: ```suggestion
To generate the Swagger UI:

- [`swagger-combine`](https://www.npmjs.com/package/swagger-combine)
```

> ## Development environment setup (only for contributing)

Reviewer: do you mean contributing to the codebase?

Reviewer 2: yes, updated

> ## Development environment setup (only for contributing)

Reviewer 2: probably we don't need this here

> For development it is highly recommended to install the configured pre-commit hooks by running from the project's root folder:
> 
> ```sh
> $ pre-commit install
> $ pre-commit install --hook-type commit-msg
> ```
> 
> This operations should be run only once. 

Reviewer: ````suggestion
Install the configured pre-commit hooks by running these commands from the project's root folder:

```sh
$ pre-commit install
$ pre-commit install --hook-type commit-msg
```
````

Reviewer: I wonder if we need line 39 here? I deleted

> This operations should be run only once. 

Probably not, since no other commands or setups specify to run commands or install anything more than once. Is there a risk if the pre-commit install  commands are run more than once?

> ## Building 
> 
> To build the project, run the `make build` command from the project's root folder. The output of the build will be generated in the `build` folder.
> 
> For cross-builds use the standard `GOOS` and `GOARCH` env vars. i.e. to build for windows:

Reviewer: ```suggestion
## Building the project

To build the project, run the `make build` command from the project's root folder. The output of the build is generated in the `build` folder.

For cross-builds, use the standard `GOOS` and `GOARCH` environment variables. For example, to build the project for Windows:
```

Reviewer: so tell us what "the project" is here. And why we want to build for another OS?

Reviewer 2: we may want to build for another os to distribute the app to another party, both in a CI system of to distribute it to a 3rd party that doesn't have a build environment

```
GOOS=windows GOARCH=amd64 make build
```

> > 💡 you can also use the default `go` command to build the project, check the content of the [Makefile](./Makefile) for reference
> 
> > ⚠️ on windows the `sha256sum` command is not present and therefore the `make build` command will build successfully but will fail to compute the checksum and copy the other resources to the build folder.

Reviewer: ```suggestion
> 💡 You can also use the default `go` command to build the project. For reference, see the contents of the [Makefile](./Makefile).

> ⚠️ on Windows, the `sha256sum` command is not present so although the `make build` command builds successfully, it fails to compute the checksum and does not copy the other resources to the build folder.
```
